src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+import mate.academy.util.Validator;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userDB = userService.findByEmail(email);
+        if (!userDB.isPresent()) {
+            throw new AuthenticationException("User doesn't exist with email: " + email);
+        }
+
+        User user = userDB.get();
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (!user.getPassword().equals(hashedPassword)) {
+            throw new AuthenticationException("Incorrect password for user with email: " + email);
+        }
```

Comment:
In my opinion, it's better for readability.

```suggestion
        if (user.getPassword().equals(hashedPassword)) {
           return user;
        }
         throw new AuthenticationException("Incorrect password for user with email: " + email);
```